og/bump/tagger.py

```diff
@@ -141,6 +141,10 @@ def generate_changelog(version: str) -> None:
 
     if any(non_bot_committers):
         logger.info("Non BOT commit detected, will not modify CHANGELOG.")
+        line = [*filter(None, changelog_file.read_text().splitlines())][0]
+        if not line.removeprefix("# ") == version:
+            raise RuntimeError(f"Unexpected version in CHANGELOG. Expected {version}, got {line}.")
+                    
```

Comment:
looks good. Are you confident we won't need this check if the bot makes changes to a file, because consistency is guaranteed then?

Reply:
Consistency is guaranteed then 😌 The value for version comes from the version bump function and should always be the expected value regardless of whether version was bumped or not